claat/nodes/iframe_test.go

```diff
@@ -0,0 +1,71 @@
+package nodes
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestNewIframeNode(t *testing.T) {
+	tests := []struct {
+		name string
+		in   string
+		out  *IframeNode
+	}{
+		{
+			name: "Empty",
+			out: &IframeNode{
+				node: node{typ: NodeIframe},
+			},
+		},
+		{
+			name: "Simple",
+			in:   "google.com",
+			out: &IframeNode{
+				node: node{typ: NodeIframe},
+				URL:  "google.com",
+			},
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			out := NewIframeNode(tc.in)
+			if diff := cmp.Diff(tc.out, out, cmp.AllowUnexported(IframeNode{}, node{})); diff != "" {
+				t.Errorf("NewIframeNode(%q) got diff (-want +got): %s", tc.in, diff)
+				return
+			}
+		})
+	}
+}
+
+func TestIframeNodeEmpty(t *testing.T) {
+	tests := []struct {
+		name string
+		in   *IframeNode
```

Comment:
Use the same pattern for both tests, either pass the url string in both or pass the IFrameNode in both.

Reply:
In the first test, the New function is what's being tested. In the second one, I want to avoid using New to isolate failures if New stops working.

Reply:
I'm not convinced that's desirable. `NewIframeNode` is the client interface we want to support, I think it makes sense to test with that rather than `IframeNode`'s directly.

Reply:
Good point. Done